core-plugins/src/main/java/io/cdap/plugin/batch/source/FTPBatchSource.java

```diff
@@ -212,62 +217,89 @@ public Schema getSchema() {
       return SCHEMA;
     }
 
+    /**
+     * This method extracts the password from url
+     */
     public String extractPasswordFromUrl() {
       int getLastIndexOfAtSign = path.lastIndexOf("@");
       String authentication = path.substring(0, getLastIndexOfAtSign);
       return authentication.substring(authentication.lastIndexOf(":") + 1);
     }
 
+    /**
+     * This method checks if extracted password from url has any special characters
+     */
     public boolean authContainsSpecialCharacters() {
       Pattern regularPasswordWithoutSpecialCharacters = Pattern.compile("[^A-Za-z0-9]");
```

Comment:
Nit: Hardcoded string value should be constant variable for regex.

Reply:
Replaced Hardcoded string value with a constant variable.

Reply:
This entire pattern can be constant right, shouldn't be compiled every time

Reply:
PR updated with necessary changes and unit tests.